pkgs/applications/editors/vim/plugins/utils/nvim-treesitter/update.py

```
@@ -1,66 +1,97 @@
 #!/usr/bin/env nix-shell
-#!nix-shell update-shell.nix -i python
+#!nix-shell update-shell.nix -i python3 -p "python3.withPackages(ps: with ps; [ requests])"
 
-import json
 import logging
 import os
 import subprocess
-import sys
 from concurrent.futures import ThreadPoolExecutor
 
+import requests
+
 log = logging.getLogger("vim-updater")
 
+# URL for the nurr JSON representation of tree-sitter parsers
+NURR_JSON_URL = "https://raw.githubusercontent.com/nvim-neorocks/nurr/main/tree-sitter-parsers.json"
+
+def generate_grammar(lang, parser_info):
+    """Generate grammar for a language based on the parser info"""
+    try:
+        # Some parsers might have a different structure
+        if "install_info" not in parser_info:
+            log.warning(f"Parser {lang} does not have install_info, skipping")
+            return ""
+
+        install_info = parser_info["install_info"]
 
-def generate_grammar(lang, rev, cfg):
-    """Generate grammar for a language"""
-    info = cfg["install_info"]
-    url = info["url"]
+        # Check for required fields
+        if "url" not in install_info or "revision" not in install_info:
```

Again, on unexpected input, we should absolutely not fail silently